### PR TITLE
Rest Client MP 1.2 fork from SmallRye

### PR DESCRIPTION
**Pull request by patriot1burke**
Fixed exception mappers
Fixed Async bugs

### CODE REVIEW

File: testsuite/microprofile-tck/pom.xml [hunk 1]
```diff
@@ -55,8 +55,9 @@
     </dependency>
 
     <dependency>
-      <groupId>org.wildfly</groupId>
-      <artifactId>wildfly-microprofile-config-smallrye</artifactId>
+      <groupId>io.smallrye</groupId>
+      <artifactId>smallrye-config</artifactId>
+      <version>1.3.6</version>
```
Comment (asoldano) [inline testsuite/microprofile-tck/pom.xml]: Out of curiosity, why is this needed?
In any case, we should also fix the resteasy-dependencies-bom/pom.xml accordingly.

Comment (patriot1burke) [reply testsuite/microprofile-tck/pom.xml]: Implementation of rest client depends on config :(

File: resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/BuilderResolver.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2015-2017 Red Hat, Inc, and individual contributors.
```
Comment (ronsigal) [inline resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/BuilderResolver.java]: Should we update the date?

**Comment by rsvoboda** [comment]
We should also sync the codebase in master 
e.g. SR Config version in master is 1.3.3 and in the branch it's 1.3.7, there is still wildfly.wildfly-microprofile-config-smallrye and version.microprofile.restclient is 1.0.1 (branch has 1.2.1)

**Comment by asoldano** [comment]
@rsvoboda https://github.com/resteasy/Resteasy/pull/2045
